master/buildbot/steps/source/git.py

```diff
@@ -534,14 +535,16 @@ def _syncSubmodule(self, _=None):
 
     def _updateSubmodule(self, _=None):
         if self.submodules:
-            return self._dovccmd(['submodule', 'update',
-                                  '--init', '--recursive', '--force'])
+            vccmd = ['submodule', 'update', '--init', '--recursive', '--force']
```

Comment:
On Mon, Apr 27 2015, Pierre Tardy wrote:
>> @@ -534,14 +535,16 @@ def _syncSubmodule(self, _=None):
>>  
>>      def _updateSubmodule(self, _=None):
>>          if self.submodules:
>> -            return self._dovccmd(['submodule', 'update',
>> -                                  '--init', '--recursive', '--force'])
>> +            vccmd = ['submodule', 'update', '--init', '--recursive', '--force']
>
> looks like this one is not unit tested. We would apreciate if you add
> a unit test for the submodule use case.

It is - see https://github.com/wentasah/buildbot/commit/198f44f7241d3ec72a418bd9cca0b29aab3b488d

Or do you have in mind some other test?

-Michal